extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -105,9 +110,27 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
         if (code == null) {
             return CompletableFuture.completedFuture(null);
         }
+
         CompletableFuture<SecurityIdentity> cf = new CompletableFuture<>();
+
+        URI absoluteUri = URI.create(context.request().absoluteURI());
+
+        Cookie stateCookie = removeCookie(context, STATE_COOKIE_NAME);
+        if (stateCookie != null) {
+            List<String> values = context.queryParam("state");
+            if (values == null || values.size() != 1 || !stateCookie.getValue().startsWith(values.get(0))) {
+                cf.completeExceptionally(new AuthenticationFailedException());
+                return cf;
+            } else {
+                String extraPath = stateCookie.getValue().split(COOKIE_DELIM)[1];
+                // This code does not work
+                context.reroute(absoluteUri.getRawPath() + extraPath);
```

Comment:
Isn't enough to send a redirect ?

Reply:
@pedroigor yeah, that should be enough actually, will try :-)

Reply:
done